File: backend/schemas/alerts.py
from pydantic import BaseModel
from typing import Optional


class AlertItem(BaseModel):
    etf_ticker: str
    date_from: str
    date_to: str
    change_type: str
    component_identifier: Optional[str] = None
    component_ticker: Optional[str] = None
    component_name: Optional[str] = None
    component_sector: Optional[str] = None
    weight_old: Optional[float] = None
    weight_new: Optional[float] = None
    weight_delta: Optional[float] = None
    shares_old: Optional[float] = None
    shares_new: Optional[float] = None
    detected_at: Optional[str] = None


class AlertsResponse(BaseModel):
    alerts: list[AlertItem]
    total: int
    page: int
    page_size: int
    total_pages: int


class SearchResultItem(BaseModel):
    etf_ticker: str
    etf_name: Optional[str] = None
    composition_date: str
    component_name: Optional[str] = None
    component_identifier: Optional[str] = None
    component_weight: Optional[float] = None
    component_sector: Optional[str] = None
    component_shares: Optional[float] = None


class SearchResponse(BaseModel):
    component_ticker: str
    found_in: list[SearchResultItem]
    total_etfs: int

File: frontend/src/types/alerts.ts
export interface AlertItem {
  etf_ticker: string
  date_from: string
  date_to: string
  change_type: string
  component_identifier: string | null
  component_ticker: string | null
  component_name: string | null
  component_sector: string | null
  weight_old: number | null
  weight_new: number | null
  weight_delta: number | null
  shares_old: number | null
  shares_new: number | null
  detected_at: string | null
}

export interface AlertsResponse {
  alerts: AlertItem[]
  total: number
  page: number
  page_size: number
  total_pages: number
}

export interface SearchResultItem {
  etf_ticker: string
  etf_name: string | null
  composition_date: string
  component_name: string | null
  component_identifier: string | null
  component_weight: number | null
  component_sector: string | null
  component_shares: number | null
}

export interface SearchResponse {
  component_ticker: string
  found_in: SearchResultItem[]
  total_etfs: number
}
